### PR TITLE
Nested provider

### PR DESCRIPTION
**Pull request by janhalama**
## Description

service-client returns invalid provider format (extended with url property). ONESDK added more strict validation of provider definition. ONESDK provider validation started to fail.

Meeting notes: https://www.notion.so/superface/Nested-provider-in-service-client-327dd338e51e4b5db4ff86427b67b089

## Motivation and Context

Deliver valid provider format for ONESDK.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by freaz** [comment]
I was thinking, that it could be split to two functions:

```ts
type ProviderMetadata = {
  provider_id: string;
  url: string;
  published_at: Date;
  published_by: string;
  owner: string;
  owner_url: string;
};

// probably not used now, but to be consistent with rest of the methods
getProvider(): ProviderMetadata;

// where provider definition is needed this would be used
getProviderDefinition(): ProviderJson;
```

Don't take it as it should be this way. This is just an idea. 

I think @janhalama @Jakub-Vacek @kysely you should agree on it.

**Comment by Jakub-Vacek** [comment]
From CLI standpoint splitting would make a lot of sense:)

**Comment by janhalama** [comment]
Backend API will return whole document (metadata + provider json).

I agree with @kysely not to add complexity (be transparent with backend api) and return whole document.

File: src/client.spec.ts [hunk 1]
```diff
@@ -842,14 +861,54 @@ describe('client', () => {
...
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    });
+
+    it('should get provider in nested format', async () => {
+      const mockResult = {
+        url: 'testUrl',
```
Comment (kysely) [inline src/client.spec.ts:900]: Minor thing, but I think provider id is missing here

```suggestion
        provider_id: 'testName',
        url: 'testUrl',
```

File: src/interfaces/providers_api_response.ts [hunk 2]
```diff
@@ -1,45 +1,11 @@
...
-  services: Array<{
-    id: string;
-    baseUrl: string;
-  }>;
-  defaultService: string;
-  securitySchemes?: ProviderSecurityScheme[];
+  owner?: string;
+  owner_url?: string;
+  published_at?: Date;
+  published_by?: string;
```
Comment (kysely) [inline src/interfaces/providers_api_response.ts:7]: Do you plan on making these required once the backend deploys the modified endpoints?

**Review by kysely — APPROVED** [review]
I have 2 minor comments but otherwise looks fine 👍

**Comment by Jakub-Vacek** [comment]
@janhalama not terribly happy with some of lint ignores but that's only temporary solution. You can merge it :)

**Pull request merged**